uboot-mediatek: update to U-Boot release v2024.07

DragonBluep opened the pull request:
**Build test only!**

DragonBluep commented on package/boot/uboot-mediatek/Makefile:
```diff
@@ -776,6 +776,10 @@ UBOOT_TARGETS := \
 
+UBOOT_CUSTOMIZE_CONFIG := \
+	--disable TOOLS_KWBIMAGE \
+	--disable TOOLS_LIBCRYPTO
```
I am not confident in this change. It seems that the original patch `no-kwbimage.patch
` was intended to avoid using OpenSSL.

dangowrt replied:
I had all this ready in my local tree already and tested on most MTK chips and devices I got here. I also removed `no-kwbimage.patch` in favor of using the now available config options.
Another thing which was annoying was the bootmenu now adds some UEFI boot entries which won't work, and one had to add `# CONFIG_CMD_BOOTEFI_BOOTMGR is not set` for all boards for it to be as before.
Please include that change as well, so existing bootmenus won't be polluted with (anyway non-working) UEFI stuff.

DragonBluep replied:
Thanks! Fixed.

BTW, the newly added `# CONFIG_CMD_BOOTEFI_BOOTMGR is not set` will be automatically removed for some devices in the second refresh commit because they don't have some `EFI` dependencies.
```
mt7622_ubnt_unifi-6-lr-v1_defconfig
mt7622_ubnt_unifi-6-lr-v2_defconfig
mt7622_ubnt_unifi-6-lr-v3_defconfig
ravpower-rp-wd009-ram_defconfig
mt7986_xiaomi_redmi-ax6000_defconfig
mt7981_xiaomi_mi-router-ax3000t_defconfig
mt7981_nokia_ea0326gmp_defconfig
mt7622_xiaomi_redmi-router-ax6s-ubi-loader_defconfig
```

dangowrt replied:
Yes, makes sense, only device which got `CONFIG_CMD_BOOTEFI` enabled should require the line to be added. And having `bootefi` command is completely optional and not required for OpenWrt. It can still be useful, e.g. to run Memtest86+, so on devices with enough space it makes sense to keep it.
However, `CONFIG_CMD_BOOTEFI_BOOTMGR` doesn't make sense unless we'd want to boot via UEFI by default (and that would imho be a bad idea, as UEFI doesn't provide any meaningful way to implement dual-boot while avoiding any single point of failure).

DragonBluep committed:
uboot-mediatek: update to U-Boot release v2024.07

1. Rename function _do_env_set() to env_do_env_set().
2. Replace kwbimage hack with UBOOT_CUSTOMIZE_CONFIG:
   "--disable TOOLS_KWBIMAGE" and "--disable TOOLS_LIBCRYPTO".
3. Disable CONFIG_CMD_BOOTEFI_BOOTMGR for all supported devices
   because the newly added UEFI bootmenu entries doesn't work.
4. Enable CONFIG_VERSION_VARIABLE for the OpenWrt One.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Co-authored-by: Daniel Golle <daniel@makrotopia.org>

DragonBluep committed:
uboot-mediatek: refresh device defconfig files

It seems that most of them are manually modified. However, we can
use `make savedefconfig` to generate a clean defconfig file.

Refreshed by:
```
Boards=(
	mt7623n_bpir2_defconfig \
	mt7623a_unielec_u7623_02_defconfig \
	mt7622_bananapi_bpi-r64-sdmmc_defconfig \
	mt7622_bananapi_bpi-r64-emmc_defconfig \
	mt7622_bananapi_bpi-r64-snand_defconfig \
	mt7622_linksys_e8450_defconfig \
	mt7622_ubnt_unifi-6-lr-v1_defconfig \
	mt7622_ubnt_unifi-6-lr-v2_defconfig \
	mt7622_ubnt_unifi-6-lr-v3_defconfig \
	ravpower-rp-wd009-ram_defconfig \
	mt7621_zbtlink_zbt-wg3526-16m_defconfig \
	mt7986_netcore_n60_defconfig \
	mt7986a_bpi-r3-emmc_defconfig \
	mt7986a_bpi-r3-nor_defconfig \
	mt7986a_bpi-r3-sd_defconfig \
	mt7986a_bpi-r3-snand_defconfig \
	mt7986_xiaomi_redmi-ax6000_defconfig \
	mt7986_tplink_tl-xdr4288_defconfig \
	mt7986_tplink_tl-xdr6086_defconfig \
	mt7986_tplink_tl-xdr6088_defconfig \
	mt7981_qihoo-360t7_defconfig \
	mt7981_xiaomi_mi-router-wr30u_defconfig \
	mt7981_h3c_magic-nx30-pro_defconfig \
	mt7986a_glinet_gl-mt6000_defconfig \
	mt7981_cmcc_rax3000m-emmc_defconfig \
	mt7981_cmcc_rax3000m-nand_defconfig \
	mt7981_jcg_q30-pro_defconfig \
	mt7986_zyxel_ex5601-t0_defconfig \
	mt7981_xiaomi_mi-router-ax3000t_defconfig \
	mt7986a_jdcloud_re-cp-03_defconfig \
	mt7986a_bpi-r3-mini-emmc_defconfig \
	mt7986a_bpi-r3-mini-snand_defconfig \
	mt7981_nokia_ea0326gmp_defconfig \
	mt7988a_bananapi_bpi-r4-emmc_defconfig \
	mt7988a_bananapi_bpi-r4-sdmmc_defconfig \
	mt7988a_bananapi_bpi-r4-snand_defconfig \
	mt7988a_bananapi_bpi-r4-poe-emmc_defconfig \
	mt7988a_bananapi_bpi-r4-poe-sdmmc_defconfig \
	mt7988a_bananapi_bpi-r4-poe-snand_defconfig \
	mt7622_xiaomi_redmi-router-ax6s-ubi-loader_defconfig \
	mt7981_openwrt-one-nor_defconfig \
	mt7981_openwrt-one-spi-nand_defconfig \
	)

for Board in ${Boards[@]}
do
	echo "Refresh board ${Board}"
	make ${Board}
	make savedefconfig
	cat ./defconfig > ./configs/${Board}
done
```

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

dangowrt committed:
uboot-mediatek: remove hard-coded UBI volume numbers

There is no point in hard-coding the UBI volume numbers as we are
dynamically looking up the volume by volume name in all cases by now.

Remove this relict as it causes problems without being useful for
anything.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>

dangowrt commented:
Tested on:
 - [x] BananaPi BPi-R3 booting from eMMC, SPI-NAND and SPI-NOR (SD card untested, but assumed working)
 - [x] BananaPi BPi-64 booting from eMMC and SNFI (SD card untested, but assumed working)
 - [x] Linksys E8450 booting from SNFI
 - [x] Ubiquity UniFi 6 LR (v1) booting from SPI-NOR
 - [x] OpenWrt One booting from SPI-NAND and SPI-NOR
 - [ ] BananaPi R2 booting from SD card and eMMC
 - [ ] BananaPi R4 booting from SD card, eMMC and SPI-NAND
 - [ ] TP-LINK TL-XDR6086 booting from SPI-NAND
